zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/TenantAttributeHolder.java

```diff
@@ -7,23 +7,41 @@
  */
 package io.camunda.zeebe.gateway.rest;
 
-import io.camunda.zeebe.protocol.record.value.TenantOwned;
+import static io.camunda.zeebe.protocol.record.value.TenantOwned.DEFAULT_TENANT_IDENTIFIER;
+import static org.springframework.web.context.request.RequestAttributes.SCOPE_REQUEST;
+
 import java.util.List;
-import org.springframework.web.server.ServerWebExchange;
+import java.util.Optional;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
 
 public final class TenantAttributeHolder {
   private static final String ATTRIBUTE_KEY = "io.camunda.zeebe.gateway.rest.tenantIds";
 
   private TenantAttributeHolder() {}
 
-  public static List<String> tenantIds(final ServerWebExchange exchange) {
-    return exchange.getAttributeOrDefault(
-        ATTRIBUTE_KEY, List.of(TenantOwned.DEFAULT_TENANT_IDENTIFIER));
+  public static List<String> tenantIds() {
+    final var requestAttributes = getCurrentRequestAttributes();
+    return getTenantsOrDefaultTenant(requestAttributes);
+  }
+
+  public static void withTenantIds(final List<String> tenantIds) {
+    final var requestAttributes = getCurrentRequestAttributes();
+    final var tenantsOptional = Optional.ofNullable(tenantIds);
+    requestAttributes.setAttribute(ATTRIBUTE_KEY, tenantsOptional, SCOPE_REQUEST);
```

Comment:
Yes, that's expected. At least, I would not know what to return. The `ServerWebExchange` was returned with the reactive stack, but this instance/class is not present in the servlet stack. So, for now, I think that's okay. It will most likely anyway evolve over time once Identity starts to build out this authentication layer further.